ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```diff
@@ -400,62 +547,15 @@ instance PraosCrypto c => ConsensusProtocol (Praos c) where
   -- (Standard) Praos uses the standard chain selection rule, so no need to
   -- override (though see note regarding clock skew).
 
-instance PraosCrypto c => NoThunks (ConsensusConfig (Praos c))
-  -- use generic instance
-
-slotEpoch :: ConsensusConfig (Praos c) -> SlotNo -> EpochNo
-slotEpoch PraosConfig{..} s =
-    runIdentity $ epochInfoEpoch epochInfo s
-  where
-    epochInfo = fixedSizeEpochInfo (EpochSize praosSlotsPerEpoch)
-    PraosParams{..} = praosParams
-
-blockInfoEpoch :: ConsensusConfig (Praos c) -> BlockInfo c -> EpochNo
-blockInfoEpoch l = slotEpoch l . biSlot
-
-epochFirst :: ConsensusConfig (Praos c) -> EpochNo -> SlotNo
-epochFirst PraosConfig{..} e =
-    runIdentity $ epochInfoFirst epochInfo e
-  where
-    epochInfo = fixedSizeEpochInfo (EpochSize praosSlotsPerEpoch)
-    PraosParams{..} = praosParams
-
-infosSlice :: SlotNo -> SlotNo -> [BlockInfo c] -> [BlockInfo c]
-infosSlice from to xs = takeWhile (\b -> biSlot b >= from)
-                      $ dropWhile (\b -> biSlot b > to) xs
-
-infosEta :: forall c. (PraosCrypto c)
-         => ConsensusConfig (Praos c)
-         -> [BlockInfo c]
-         -> EpochNo
-         -> Natural
-infosEta l _  0 = praosInitialEta l
-infosEta l xs e =
-    let e'   = e - 1
-        eta' = infosEta l xs e'
-        from = epochFirst l e'
-        n    = div (2 * praosSlotsPerEpoch) 3
-        to   = SlotNo $ unSlotNo from + n
-        rhos = reverse [biRho b | b <- infosSlice from to xs]
-    in  bytesToNatural
-          . hashToBytes
-          $ hashWithSerialiser @(PraosHash c) toCBOR (eta', e, rhos)
-  where
-    PraosParams{..} = praosParams l
-
-infosStake :: ConsensusConfig (Praos c) -> [BlockInfo c] -> EpochNo -> StakeDist
-infosStake s@PraosConfig{..} xs e = case ys of
-    []                  -> praosInitialStake
-    (BlockInfo{..} : _) -> biStake
-  where
-    e' = if e >= 2 then EpochNo (unEpochNo e - 2) else 0
-    ys = dropWhile (\b -> blockInfoEpoch s b > e') xs
-
+-- | Probability for stakeholder Uᵢ to be elected in slot
+-- slⱼ considering its relative stake αᵢ.
 phi :: ConsensusConfig (Praos c) -> Rational -> Double
-phi PraosConfig{..} r = 1 - (1 - praosLeaderF) ** fromRational r
+phi PraosConfig{..} alpha = 1 - (1 - praosLeaderF) ** fromRational alpha
   where
     PraosParams{..} = praosParams
 
+-- | Compute Tᵢ for a given stakeholder @n@ at a @SlotNo@. Will be computed from
+-- @praosEvolvingStake@.
```

Comment:
Let's change the `@` to `'`: `'praosEvolvingStake'`

Reply:
Applied.